lib/core/filtering/simple_rules/validation/no_conflicting_policies_rules_validator.rb

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+# frozen_string_literal: true
+
+require 'core/filtering/simple_rules/validation/simple_rules_validator'
+require 'core/filtering/validation_status'
+
+module Core
+  module Filtering
+    module SimpleRules
+      module Validation
+        class NoConflictingPoliciesRulesValidator < Core::Filtering::SimpleRules::Validation::SimpleRulesValidator
+
+          def are_rules_valid
+            rule_field_value_to_policy = {}
+
+            @rules.each do |simple_rule|
+              rule_field_value = simple_rule.slice('rule', 'field', 'value')
+              policy = simple_rule['policy']
+
+              return conflicting_rules(rule_field_value) if rule_field_value_to_policy.key?(rule_field_value) && rule_field_value_to_policy[rule_field_value]['policy'] != policy
```

Comment:
```suggestion
              return conflicting_rules(rule_field_value) if rule_field_value_to_policy.key?(rule_field_value)
```

Should we ever have the same `rule_field_value`, even if the policy is the same? I think the combo of those 3 should be unique, right?